### PR TITLE
Feature following and followers tables

### PR DESCRIPTION
**Pull request by amittes**

### CODE REVIEW

File: client/src/components/userProfile/Followers.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    const loadRequestFn = async (currentPage) => {
+        const response = await followsService.loadFollowers(currentUser.email, currentPage);
+        return {
+            totalElements: response.totalAmount,
+            content: response.data
+          };
+    };
+
+    const removeRequestFn = async(user) => {
+       await followsService.updateFollower("REMOVE", user.email_from, user.email_to);
```
Comment (AviranBac) [inline client/src/components/userProfile/Followers.js:23]: `"REMOVE"` -> `ActionType.REMOVE`

File: server/src/services/follows.js [hunk 2]
```diff
@@ -1,13 +1,23 @@
 const Follows = require("../db/mongo/models/follow");
 
-const searchFollowings = async (email) => {
-    return Follows.find({ email_from: email}).exec();
+const searchFollowings = async (email, pageNumber, pageLimit) => {
+    const totalAmount = (await Follows.find({ email_from: email}).exec()).length;
```
Comment (AviranBac) [inline server/src/services/follows.js:4]:
```
const aggregationQuery = [  
    { $match: { email_from: email }} 
];
const data = await Follows.aggregate([...aggregationQuery, ...paginationQuery(pageNumber, pageLimit)]);
const dataTotalAmount = await Follows.aggregate([...aggregationQuery, { $count: "count" }]);
const totalAmount = dataTotalAmount[0] ? dataTotalAmount[0].count : 0;

return { data, totalAmount };
```
Similar to what Shuker did in series. Apply something similar in `searchFollowers`

File: client/src/services/follows.service.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        params: { pageNumber }
+    }).then(response => response.data)
+        .catch(error => {
+            console.error(`Error while trying to fetch most followings details. Error: ${error}`);
+            throw error;
+        });
+};
+
+const updateFollower = (action, emailFrom, emailTo) => {
+    return axios.post(`${config.serverUrl}/follows`, { action, emailFrom, emailTo })
```
Comment (AviranBac) [inline client/src/services/follows.service.js:25]: This method will work for both add and remove. Update `console.error` so it won't relate to "remove" only.

Comment (AviranBac) [inline client/src/services/follows.service.js:25]: What's the difference between `updateFollower` and `updateFollowing` in functionality?
You can basically do `updateFollow` which will do exactly the same, and you'll send the correct parameters based on the situation.

File: client/src/components/userProfile/Followings.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    const loadRequestFn = async (currentPage) => {
+        const response = await followsService.loadFollowings(currentUser.email, currentPage);
+        return {
+            totalElements: response.totalAmount,
+            content: response.data
+          };
+    };
+
+    const removeRequestFn = async(user) => {
+       await followsService.updateFollowing("REMOVE", user.email_from, user.email_to);
```
Comment (AviranBac) [inline client/src/components/userProfile/Followings.js:23]: `"REMOVE"` -> `ActionType.REMOVE`

File: client/src/components/userProfile/Followings.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    const loadRequestFn = async (currentPage) => {
+        const response = await followsService.loadFollowings(currentUser.email, currentPage);
+        return {
+            totalElements: response.totalAmount,
+            content: response.data
+          };
+    };
+
+    const removeRequestFn = async(user) => {
```
Comment (AviranBac) [inline client/src/components/userProfile/Followings.js:22]: `async (follow)`

File: client/src/components/userProfile/Followers.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    };
+
+
+    const routerLinkExtractor = (user) => {
```
Comment (AviranBac) [inline client/src/components/userProfile/Followers.js:27]:
```
const routerLinkExtractor = (follow) => {
        return `/user/${follow.email_from}`
};

File: client/src/components/userProfile/Followings.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+       await followsService.updateFollowing("REMOVE", user.email_from, user.email_to);
+    };
+
+
+    const routerLinkExtractor = (user) => {
```
Comment (AviranBac) [inline client/src/components/userProfile/Followings.js:27]:
```
const routerLinkExtractor = (follow) => {
        return `/user/${follow.email_to}`
};

File: client/src/Routing.js [hunk 8]
```diff
@@ -20,6 +22,12 @@ const Routing = () => {
             <Route path="/statistics" element={<AuthenticationGuardRoute />}>
                 <Route path="/statistics" element={<Statistics />} />
             </Route>
+            <Route path="/profile/followers" element={<AuthenticationGuardRoute />}>
```
Comment (AviranBac) [inline client/src/Routing.js:25]: Remember to update when Shuker adds tabbar if she merges before you
Also, it should have been `/user/:email/followers` and `/user/:email/followings`

File: client/src/components/userProfile/Followers.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+    const loadRequestFn = async (currentPage) => {
+        const response = await followsService.loadFollowers(currentUser.email, currentPage);
+        return {
+            totalElements: response.totalAmount,
+            content: response.data
+          };
+    };
+
+    const removeRequestFn = async(user) => {
```
Comment (AviranBac) [inline client/src/components/userProfile/Followers.js:22]: Should you be able to remove those who follow you? It'll ruin their "recommendations"
In my original project they couldn't remove in this section, think about it

**Review by AviranBac — CHANGES_REQUESTED** [review]

File: client/src/Routing.js (hunk 8 above)
Comment (amittes) [reply client/src/Routing.js:25]: אם אתה מתכוון בפרונט סבבה, 
בבקאנד זה מוגדר עם follows ולא user

File: client/src/components/userProfile/Followers.js (hunk 9 above)
Comment (amittes) [reply client/src/components/userProfile/Followers.js:22]: זה כמו שבאינסטגרם אתה יכול להוריד אנשים שעוקבים אחריך, אני יכולה להוריד את זה

File: client/src/Routing.js (hunk 8 above)
Comment (AviranBac) [reply client/src/Routing.js:25]: כן הכוונה ל-route בפרונט

File: client/src/components/userProfile/Followers.js (hunk 9 above)
Comment (AviranBac) [reply client/src/components/userProfile/Followers.js:22]: קיבלתי

File: client/src/components/userProfile/Followers.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
+import PaginationTable from '../paginationTable/PaginationTable';
```
Comment (AviranBac) [inline client/src/components/userProfile/Followers.js:1]: Pull from main

**Review by AviranBac — CHANGES_REQUESTED** [review]

**Review by AviranBac — APPROVED** [review]

**Pull request merged**